core/model/modx/moddashboardwidget.class.php

```diff
@@ -317,8 +317,9 @@ public function getFileChunk($tpl,array $placeholders = array()) {
     public function renderAsSnippet($content = '') {
         if (empty($content)) $content = $this->widget->get('content');
         $content = str_replace(array('<?php','?>'),'',$content);
-        $closure = create_function('$scriptProperties','global $modx;if (is_array($scriptProperties)) {extract($scriptProperties, EXTR_SKIP);}'.$content);
-        return $closure(array(
+        $snippet = $this->modx->newObject('modSnippet');

```

Comment:
I think it would be good to set the name to something like `md5($this->widget->get('content'))` or similar.